File: Add Minimum Characters - GFG/add-minimum-characters.py
#User function Template for python3

class Solution:
    def addMinChar (self, S):
        # code here 
        n = len(S)
        i,j,ans = 0,n-1,n-1
        
        while i<=j:
            if S[i]==S[j]:
                i+=1
                j-=1
            else:
                ans-=1
                i=0
                j = ans
                
        return n-ans-1

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        str1 = input()

        ob = Solution()
        print(ob.addMinChar(str1))

# } Driver Code Ends

File: Add Minimum Characters - GFG/add-minimum-characters.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++

class Solution{   
public:
    int addMinChar(string str){    
        int n=str.size();
        int i=0, j=n-1;
        int ans=n-1;
        while(i<=j){
            if(str[i]==str[j]){
                i++;
                j--;
            }
            else {
                ans--;
                i=0;
                j=ans;
            }
        }
        return n-ans-1;
    }
};

//{ Driver Code Starts.


int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        string str;
        cin >> str;
        
        Solution ob;
        cout << ob.addMinChar(str) << endl;
    }
    return 0; 
} 
// } Driver Code Ends